ontotext-yasgui-web-component/src/components/ontotext-yasgui-web-component/ontotext-yasgui-web-component.tsx

```diff
@@ -620,6 +628,59 @@ export class OntotextYasguiWebComponent {
     this.showShareQueryDialog = true;
   }
 
+  /**
+   * Handler for the event fired when the AI explain query button in the editor is triggered.
+   */
+  @Listen('internalExplainQueryEvent')
+  explainQueryHandler() {
+    this.getOntotextYasgui()
+      .then((ontotextYasgui) => {
+        ontotextYasgui.query(undefined, EXPLAIN_PLAN_TYPE.LLM_EXPLAIN).catch(() => {
+          // catch this to avoid unhandled rejection
+        });
+      });
+    this.showShareQueryDialog = false;
+  }
+
+  /**
+   * Handler for the event fired when an action is selected from the yasqe dropdown menu.
+   * @param event The event containing the action to be performed.
+   */
+  @Listen('internalYasqeDropdownActionSelected')
+  onYasqeDropdownActionSelected(event: CustomEvent<{ payload: string }>) {
+    const action = event.detail.payload;
+    let userQuery = '';
+    this.getOntotextYasgui().then(ontotextYasgui => {
+      switch (action) {
+        case 'explain_plan':
+          userQuery = ontotextYasgui.getQuery();
+          ontotextYasgui.setQuery(this.removeQueryLLMComments(userQuery));
+          ontotextYasgui.query(undefined, EXPLAIN_PLAN_TYPE.EXPLAIN).catch(() => {
+            // catch this to avoid unhandled rejection
+          });
+          break;
+        case 'explain_all':
+          userQuery = ontotextYasgui.getQuery();
+          ontotextYasgui.setQuery(this.removeQueryLLMComments(userQuery));
+          ontotextYasgui.query(undefined, EXPLAIN_PLAN_TYPE.LLM_EXPLAIN).catch(() => {
+            // catch this to avoid unhandled rejection
+          });
+          break;
+        case 'explain_query':
+        case 'explain_results': {
+          const mode = action === 'explain_query' ? 'query' : 'results';
```

Comment:
IMO, it'd be better if you just implement both cases explain_query and explain_results instead of falling through and doing conditional logic here. There will be some duplication but that's not a big deal in this case.